test/io.carp

```diff
@@ -1,6 +1,16 @@
 (load "Test.carp")
 (use-all IO Test)
 
+(defn write-then-read [content file-name]
+  (let [ written? (IO.write-file content file-name) ]
+    (if (Result.error? &written?)
+      (Result.unsafe-from-error written?)
+      (let-do [ read? (IO.read-file file-name) ]
+        (IO.unlink @file-name)
```

Comment:
a little scary to use this in a test...

Reply:
maybe just read the file and check its contents

Reply:
The basic idea was to show that the round-trip works (e.g. writing and reading to/from file does not change anything).

And the test works fine, even on Windows.
Would you still like me to reduce it to reading the file?